src/main/scala/mesosphere/marathon/core/task/termination/impl/KillServiceActor.scala

```diff
@@ -128,25 +128,32 @@ private[impl] class KillServiceActor(
 
   def processKill(toKill: ToKill): Unit = {
     val instanceId = toKill.instanceId
-    val taskId = toKill.taskIdToKill
+    val taskIds = toKill.taskIdsToKill
 
     // TODO(PODS): align this with other Terminal/Unreachable/whatever extractors
-    val taskIsLost: Boolean = toKill.maybeInstance.fold(false) { instance =>
+    val isLost: Boolean = toKill.maybeInstance.fold(false) { instance =>
       instance.isGone || instance.isUnknown || instance.isDropped || instance.isUnreachable
     }
 
-    if (taskIsLost) {
-      log.warning("Expunging lost {} from state because it should be killed", taskId)
-      // we will eventually be notified of a taskStatusUpdate after the task has been expunged
+    // An instance will be expunged once all tasks are terminal. Therefore, this case is
+    // highly unlikely. Should it ever occur, this will still expunge the instance to clean up.
+    val allTerminal: Boolean = taskIds.isEmpty
+
+    if (isLost || allTerminal) {
+      val msg = if (isLost) "it is lost" else "all its tasks are terminal"
+      log.warning("Expunging {} from state because {}", instanceId, msg)
```

Comment:
We probably should use scala logging (StrictLogging) which implements log as a macro that does "isLevelEnabled..."